src/host/output.cpp

```diff
@@ -512,13 +512,9 @@ void CloseConsoleProcessState()
 
     HandleCtrlEvent(CTRL_CLOSE_EVENT);
 
-    // Jiggle the handle: (see MSFT:19419231)
-    // When we call this function, we'll only actually close the console once
-    //      we're totally unlocked. If our caller has the console locked, great,
-    //      we'll dispatch the ctrl event once they unlock. However, if they're
-    //      not running under lock (eg PtySignalInputThread::_GetData), then the
-    //      ctrl event will never actually get dispatched.
-    // So, lock and unlock here, to make sure the ctrl event gets handled.
+    // MSFT:19419231
+    // Ensure ctrl events are processed before exiting,
+    // no matter whether the console is locked or not.

```

Comment:
May want to add an archeological 
```suggestion
    // no matter whether the console is locked or not.
    // GH#12181: we used to "jiggle the handle" here to trigger ProcessCtrlEvents in the scenario where we were called outside of lock (e.g. PtySignalInputThread::_GetData). WE no longer need to worry about that.....
```